include/openenclave/corelibc/arpa/inet.h

```diff
@@ -0,0 +1,109 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* Licensed under the MIT License. */
+
+#ifndef _OE_ARPA_INET_H
+#define _OE_ARPA_INET_H
+
+#include <openenclave/bits/defs.h>
+#include <openenclave/bits/types.h>
+#include <openenclave/corelibc/netinet/in.h>
+
+OE_EXTERNC_BEGIN
+
+/*
+**==============================================================================
+**
+** OE names:
+**
+**==============================================================================
+*/
+
+#define __OE_LITTLE_ENDIAN 1234
+#define __OE_BIG_ENDIAN 4321
+
+#if defined(__i386) || defined(__x86_64)
+#define __OE_BYTE_ORDER __OE_LITTLE_ENDIAN
+#elif defined(__arm__) || defined(__aarch64__)
+#define __OE_BYTE_ORDER __OE_BIG_ENDIAN
+#elif defined(_WIN32)
+#define __OE_BYTE_ORDER __OE_LITTLE_ENDIAN
+#endif
+
+OE_INLINE uint32_t oe_ntohl(uint32_t x)
+{
+#if __OE_BYTE_ORDER == __OE_BIG_ENDIAN
+    return x;
+#else
+    return __builtin_bswap32(x);
+#endif
+}
+
+OE_INLINE uint16_t oe_ntohs(uint16_t x)
+{
+#if __OE_BYTE_ORDER == __OE_BIG_ENDIAN
+    return x;
+#else
+    return __builtin_bswap16(x);
+#endif
+}
+
+OE_INLINE uint32_t oe_htonl(uint32_t x)
+{
+#if __OE_BYTE_ORDER == __OE_BIG_ENDIAN
+    return x;
+#else
+    return __builtin_bswap32(x);
+#endif
+}
+
+OE_INLINE uint16_t oe_htons(uint16_t x)
+{
+#if __OE_BYTE_ORDER == __OE_BIG_ENDIAN
+    return x;
+#else
+    return __builtin_bswap16(x);
+#endif
+}
+
+uint32_t oe_inet_addr(const char* cp);
+
+/*
+**==============================================================================
+**
+** Standard-C names:
+**
+**==============================================================================
+*/
+
+#if defined(OE_NEED_STDC_NAMES)
+
+OE_INLINE uint32_t ntohl(uint32_t x)
+{
+    return oe_ntohl(x);
+}
+
+OE_INLINE uint16_t ntohs(uint16_t x)
+{
+    return oe_ntohs(x);
+}
+
+OE_INLINE uint32_t htonl(uint32_t x)
+{
+    return oe_htonl(x);
+}
+
+OE_INLINE uint16_t htons(uint16_t x)
```

Comment:
Just for my understanding. Why do these need to be inline functions and not 
#define htos oe_htons
?

Reply:
# define htons             oe_htons
is how it appears in the new_platforms branch, so I have the same question

Reply:
In this case macros would work fine. In other cases where function parameters need to be cast to oe-prefixed types, inlines work better. This has been our convention before this PR.

Consider this case.

```
OE_INLINE int stat(const char* pathname, struct stat* buf)
{
    return oe_stat(pathname, (struct oe_stat*)buf);
}
```

One could replace this with:

```
#define stat oe_stat
```

But this is subtle, because "stat" is both a function name and a structure name. Worse, this substitution propagates into user code, like this.

```
struct stat stat; /* stat renamed to oe_stat. If this variable is not used, the compiler warning will read "oe_stat is unused". */
```

Also, you have to worry about user code that redefines "stat" after including the corelibc header.

Finally, the macro approach does not play well with C++ function overloading. Consider the following.

```
int oe_foo(int x);
#define foo oe_foo
.
.
.
/* User defined functions. */
int oe_foo(int x, int y)
{
    ...
}
int foo(int x, int y)
{
    ...
}
```

This results in a "oe_foo" already defined error.